File: LeetCode/cpp/2022_Convert_1D_Array_Into_2D_Array.cpp
class Solution {
public:
	vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
		vector<vector<int>> answer;

		if (m * n != original.size()) {
			return answer;
		}

		answer.assign(m, vector<int>(n));

		int idx = 0;
		for (int i = 0; i < answer.size(); i++) {
			for (int j = 0; j < answer[0].size(); j++) {
				answer[i][j] = original[idx++];
			}
		}

		return answer;
	}
};

File: LeetCode/java/2022_Convert_1D_Array_Into_2D_Array.java
class Solution {
    public int[][] construct2DArray(int[] original, int m, int n) {
        int[][] answer = {};

        if (m * n != original.length) {
            return answer;
        }

        answer = new int[m][n];

        int idx = 0;
        for (int i = 0; i < answer.length; i++) {
            for (int j = 0; j < answer[0].length; j++) {
                answer[i][j] = original[idx++];
            }
        }

        return answer;
    }
}